tests/CMakeLists.txt

```diff
@@ -0,0 +1,210 @@
+  ${Boost_INCLUDE_DIRS}
+  ${OPAE_SDK_SOURCE}/common/include
+  ${OPAE_SDK_SOURCE}/tests)
+
+message("jsoncpp=${jsoncpp_INCLUDE_DIRS}")
+add_library(commonlib SHARED common_test.h common_test.cpp)
+target_link_libraries(commonlib opae-c ${GTEST_BOTH_LIBRARIES} ${jsoncpp_LIBRARIES})
+target_include_directories(commonlib PUBLIC
+  $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<BUILD_INTERFACE:${jsoncpp_INCLUDE_DIRS}>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE $<BUILD_INTERFACE:${OPAE_SDK_SOURCE}/libopae/src>)
+
+add_executable(foapp
+  fpgaopen_app.cpp)
+target_include_directories(foapp PUBLIC
+  $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE $<BUILD_INTERFACE:${OPAE_SDK_SOURCE}/libopae/src>)
+target_link_libraries(foapp commonlib safestr opae-c uuid ${GTEST_BOTH_LIBRARIES})
+
+set(SRC gtmain.cpp
+  jsonParser.cpp
+  unit/gtProperties_base.cpp
+  unit/gtOpenClose_base.cpp
+  unit/gtEnumerate.cpp
+  unit/gtOpen.cpp
+  unit/gtOptionParser.cpp
+  function/gtEnumerate.cpp
+  function/gtHostif.cpp
+  function/gtOpenClose.cpp
+  function/gtReconf.cpp
+  function/gtUmsg.cpp
+  function/gtBuffer.cpp
+  function/gtEvent.cpp
+  function/gtMMIO.cpp
+  function/gtReset.cpp
+  function/gtMMLink.cpp
+  stress/gtBuf_stress.cpp
+  stress/gtMMIO_stress.cpp
+  stress/gtOpen_stress.cpp
+  stress/gtPR_stress.cpp
+  stress/gtPerfCounters_stress.cpp
+  stress/gtRASPower_stress.cpp)
+add_executable(gtapi ${SRC})
+target_include_directories(gtapi PUBLIC
+  $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE $<BUILD_INTERFACE:${OPAE_SDK_SOURCE}/libopae/src>)
+target_link_libraries(gtapi commonlib safestr opae-c json-c uuid ${GTEST_BOTH_LIBRARIES}
+  opae-c++-utils opae-c++)
+configure_file(${PROJECT_SOURCE_DIR}/configuration.json
+  ${CMAKE_BINARY_DIR}/tests/configuration.json
+  COPYONLY)
+
+if(THREADS_HAVE_PTHREAD_ARG)
+  target_compile_options(PUBLIC commonlib "-pthread")
+  target_compile_options(PUBLIC foapp "-pthread")
+  target_compile_options(PUBLIC gtapi "-pthread")
+endif()
+
+if(CMAKE_THREAD_LIBS_INIT)
+  target_link_libraries(foapp "${CMAKE_THREAD_LIBS_INIT}")
+  target_link_libraries(gtapi "${CMAKE_THREAD_LIBS_INIT}")
+endif()
+
+# build mock driver fake directory structure
+add_custom_target(mock-sysfs-prepare)
+add_custom_command(TARGET mock-sysfs-prepare POST_BUILD
+  COMMAND cmake -E copy ${CMAKE_CURRENT_SOURCE_DIR}/mock_sys_tmp-1socket-nlb0.tar.gz /tmp
+  COMMAND tar xzvf /tmp/mock_sys_tmp-1socket-nlb0.tar.gz -C /tmp --strip 1)
+
+# build mock driver
+add_library(mock SHARED mock.c)
+target_include_directories(mock PUBLIC
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  PRIVATE $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/libopae/src>)
+add_dependencies(mock mock-sysfs-prepare)
+target_link_libraries(mock dl safestr)
+
+# Add discrete tests
+add_subdirectory(dcp)
```

Comment:
Please don't include dcp directory  in the tests/ folder, this is a temporary folder that  DCP team wants to keep their tests.